packages/react-devtools-scheduling-profiler/src/content-views/utils/text.js

```diff
@@ -32,14 +32,34 @@ export function trimText(
   text: string,
   width: number,
 ): string | null {
-  for (let i = text.length - 1; i >= 0; i--) {
-    const trimmedText = i === text.length - 1 ? text : text.substr(0, i) + '…';
+  const maxIndex = text.length - 1;
+
+  let startIndex = 0;
+  let stopIndex = maxIndex;
+
+  let longestValidIndex = 0;
+  let longestValidText = null;
+
+  // Trimming long text could be really slow if we decrease only 1 character at a time.
+  // Trimming with more of a binary search approach is faster in the worst cases.
+  while (startIndex <= stopIndex) {
```

Comment:
just curious, do you know ballpark how much slower was this before / how much faster is it now? did we run into cases with long strings a lot? or is this a case that becomes more prevalent with the network measures?

Reply:
The network measures have URLs that are sometimes _really really_ long. This change was meant to kind of fix the "worst case" scenario, although I also pre-trimmed long URL strings too. Committed this as kind of a just-in-case backup plan.